src/vulkan/buffer.cc

```diff
@@ -103,47 +87,12 @@ void Buffer::Shutdown() {
     vkDestroyBufferView(GetDevice(), view_, nullptr);
 
   if (memory_ != VK_NULL_HANDLE) {
-    if (is_buffer_host_accessible_)
-      UnMapMemory(memory_);
-
+    UnMapMemory(memory_);
     vkFreeMemory(GetDevice(), memory_, nullptr);
   }
 
   if (buffer_ != VK_NULL_HANDLE)
     vkDestroyBuffer(GetDevice(), buffer_, nullptr);
-
-  if (!is_buffer_host_accessible_)
-    Resource::Shutdown();
```

Comment:
Do we call Resource::Shutdown elsewwhere?

Reply:
Now `Resource::Shutdown()` does not do anything because now, with this CL, both `host_accessible_memory_` and `host_accessible_buffer_` are `VK_NULL_HANDLE`.

However, to prevent potential bugs later e.g., we update `Resource::Shutdown()` in the future, I put it here.